src/django_components/dependencies.py

```diff
@@ -183,31 +270,58 @@ def cache_inlined_css(comp_cls: Type["Component"], content: str) -> None:
 #########################################################
 
 
-class Dependencies(NamedTuple):
-    # NOTE: We pass around the component CLASS, so the dependencies logic is not
-    # dependent on ComponentRegistries
-    component_cls: Type["Component"]
-    component_id: str
+def _link_dependencies_with_component_html(
+    component_id: str,
+    html_content: str,
+    css_input_hash: Optional[str],
+) -> str:
+    elems = SoupNode.from_fragment(html_content)
+
+    # Insert component ID
+    for elem in elems:
+        # Ignore comments, text, doctype, etc.
+        if not elem.is_element():
+            continue
+
+        # Component ID is used for executing JS script, e.g. `data-djc-id-a1b2c3`
+        #
+        # NOTE: We use `data-djc-css-a1b2c3` and `data-djc-id-a1b2c3` instead of
+        # `data-djc-css="a1b2c3"` and `data-djc-id="a1b2c3"`, to allow
+        # multiple values to be associated with the same element, which may happen if
+        # One component renders another.
+        elem.set_attr(f"data-djc-id-{component_id}", True)
+
+        # Attribute by which we bind the CSS variables to the component's CSS,
+        # e.g. `data-djc-css-a1b2c3`
+        if css_input_hash:
```

Comment:
We now add `data-djc-id-xxxxxx` to each component, where `xxxxxx` stands for the component's render ID.

If the component defines CSS variables (and defines `Component.css`), we also insert `data-djc-css-yyyyyy` attribute to tell the browser which HTML elements should have access to the CSS variables identified by the `yyyyyy` hash.